gpytorch/lazy/lazy_tensor.py

```diff
@@ -731,6 +731,106 @@ def add_jitter(self, jitter_val=1e-3):
         diag = torch.tensor(jitter_val, dtype=self.dtype, device=self.device)
         return self.add_diag(diag)
 
+    def cat_rows(self, fant_train, fant_fant):
+        """
+        Concatenates new rows and columns to the matrix that this LazyTensor represents, e.g.
+        C = ((A; B); B', D).
+
+        TODO: add reference.
+        """
+        from .cat_lazy_tensor import CatLazyTensor
+        from .root_lazy_tensor import RootLazyTensor
+
+        batch_shape = fant_train.shape[:-2]
+        L = self.root_decomposition().root.evaluate()
+
+        L_inverse = self.root_inv_decomposition().root.evaluate()
+
+        m, n = L.shape[-2:]
+
+        lower_left = fant_train.matmul(L_inverse)
+        schur = fant_fant - lower_left.matmul(lower_left.transpose(-2, -1))
+        schur_root = schur.root_decomposition().root.evaluate()
+
+        # Form new root Z = [L 0; lower_left schur_root]
+        num_fant = schur_root.size(-2)
+        new_root = torch.zeros(*batch_shape, m + num_fant, n + num_fant, device=L.device, dtype=L.dtype)
+        new_root[..., :m, :n] = L
+        new_root[..., m:, : lower_left.shape[-1]] = lower_left
+        new_root[..., m:, n : (n + schur_root.shape[-1])] = schur_root
+
+        new_lazy_tensor_1 = CatLazyTensor(self, fant_train, dim=-2)
+        new_lazy_tensor_2 = CatLazyTensor(fant_train.transpose(-1, -2), fant_fant, dim=-2)
+        new_lazy_tensor = CatLazyTensor(new_lazy_tensor_1, new_lazy_tensor_2, dim=-1)
```

Comment:
I wonder if it makes sense to have some kind of `LazyTensorWithNewRowsAndCols` object that then internally just uses the base lazy tensor and the `fant_train` and `fant_fant` components.